lib/include/jxl/decode.h

```diff
@@ -1287,9 +1287,11 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxSizeRaw(const JxlDecoder* dec,
  *
  * @param dec decoder object
  * @param detail at which level of detail to trigger JXL_DEC_FRAME_PROGRESSION
+ * @return JXL_DEC_SUCCESS on success, JXL_DEC_ERROR on error, such as an
+ * invalid value for the progressive detail.
  */
-JXL_EXPORT void JxlDecoderSetProgressiveDetail(JxlDecoder* dec,
-                                               uint32_t detail);
+JXL_EXPORT JxlDecoderStatus JxlDecoderSetProgressiveDetail(JxlDecoder* dec,
+                                                           uint32_t detail);
```

Comment:
One question is if this should only affect when the progression event gets triggered, or if it should also affect what happens when a Flush is done on partial input. That way you can ignore the progression events but still make sure that e.g. only a limited amount of refreshes are done even if Flush is called multiple times.

Reply:
You don't need a new API function to do Flush only when more input is needed and a progressive event exists: you just need to remember whether you got a progressive event until the next time ProcessInput returns NEED_MORE_INPUT.

Reply:
Yes, that avoids doing multiple flushes per arriving bitstream chunk, but it may lead to the flush showing "too much" information, e.g. part of a scan when you only want to show complete scans.